guides/v2.2/frontend-dev-guide/layouts/xml-manage.md

```diff
@@ -383,7 +382,95 @@ In this file, reference the element having added the `remove` attribute:
 
 ## Replace elements {#layout_markup_replace_elements}
 
-To replace an element, [remove it] and add a new one.
+Elements can be replaced with custom blocks by one of the following methods below:
+
+ * Remove the old `<block>` and substitute it with a new `<block>` with the same reference and path  
+ * Override the `<block>` definition through the class preference method
+
+**Example 1**: substitute the Product View attributes `<block>` of a store catalog with a new `<block>` of the same name across all content pages of the website.
+
+The following block, which is our old block, is declared in `app/design/frontend/catalog/product_view_attributes/layout/local.xml`:
+
+```xml
+<block type="catalog/product_view_attributes" name="product.attributes" as="additional" template="catalog/product/view/attributes.phtml">
+    <action method="addToParentGroup"><group>detailed_info</group></action>
+    <action method="setTitle" translate="value"><value>Additional Information</value></action>
+</block>
+```
+You can replace this block in `layout.xml` with a custom block that uses the same name, as follows:
+
+```xml
+<block type="attributesasgroup/groupview" name="product.attributes" as="additional" template="webguys/attributesasgroup/groupview.phtml">
+    <action method="addToParentGroup"><group>detailed_info</group></action>
+    <action method="setTitle" translate="value"><value>Additional Information</value></action>
+</block>
+```
+
+As a final note, it is important that your block be defined on the backend side. If it is not defined, Magento will not understand which block you are referencing to it. This will mean that Magento will not properly render the content that is located inside your block.
+
+For instance, in order for the new block in the above case to be processed properly, Magento needs `Namespace_Modulename_Block_Groupview` to be defined at `app/code/local/Namespace/Modulename/Block/Groupview.php` (where `Namespace_Modulename` stands for your custom module name).
+
+Some block definitions do not contain an explicit name attribute, which means you cannot override the block from a layout XML file. For these blocks, you can add functionality through ViewModels or modify functionality through plugins (interceptors).
+
+**Example 2**: add functionality with a ViewModel inserted into the `cart/item/default.phtml` template found in `Magento/Checkout/view/frontend/layout/checkout_cart_item_renderers.xml`.
```

Comment:
I suggest removing this from an "Example" point and creating a new section: `## Add functionality to existing elements {#layout_markup_add_to_elements}`. This section is a wonderful additional to this document!

I think this section is important enough that it could do well in its own topic.